modules/remotecontrol/index.js

```diff
@@ -64,136 +61,148 @@ class RemoteControl {
     /**
      * Initializes the remote control functionality.
      */
-    init(channel, windowManager, handleAuthorization) {
-        this.handleAuthorization = handleAuthorization;
-        this.windowManager = windowManager;
+    init(channel) {
         this.channel = channel;
         this.channel.ready(() => {
             this.channel.listen('message', message => {
-                const event = message.data;
-                if(event.name === REMOTE_CONTROL_EVENT_NAME) {
-                    this.onRemoteControlEvent(event);
+                const { name } = message.data;
+                if(name === REMOTE_CONTROL_MESSAGE_NAME) {
+                    this.onRemoteControlMessage(message);
                 }
             });
-            this.sendEvent({type: EVENT_TYPES.supported});
-            if (!handleAuthorization) {
-                this.start();
-            }
+            this.sendEvent({ type: EVENTS.supported });
         });
     }
 
     /**
      * Disposes the remote control functionality.
      */
     dispose() {
-        this.windowManager = null;
         this.channel = null;
         this.stop();
     }
 
     /**
-     * Handles permission requests from Jitsi Meet.
-     * @param {object} userInfo - information about the user that has requested
-     * permissions:
-     * @param {string} userInfo.displayName - display name
-     * @param {string} userInfo.userJID - the JID of the user.
-     * @param {string} userInfo.userId - the user id (the resource of the JID)
-     * @param {boolean} userInfo.screenSharing - true if the screen sharing
-     * is started.
+     * Handles remote control start messages.
+     *
+     * @param {number} id - the id of the request that will be used for the
+     * response.
+     * @param {string} sourceId - The source id of the desktop sharing stream.
      */
-    handlePermissionRequest(userInfo) {
-        this.windowManager.requestRemoteControlPermissions(userInfo)
-            .then(result => {
-                this.sendEvent({
-                    type: EVENT_TYPES.permissions,
-                    action: result ? PERMISSIONS_ACTIONS.grant
-                        : PERMISSIONS_ACTIONS.deny,
-                    userId: userInfo.userId
-                });
-                if(result) {
-                    this.start();
+    start(id, sourceId) {
+        const displays = electron.screen.getAllDisplays();
+
+        switch(displays.length) {
+            case 0:
+                this.display = undefined;
+            break;
+            case 1: // Linux
```

Comment:
Why is there a comment about Linux here? Is Linux the only case where 1 is returned for display length?